### ISSUE TITLE
revive can't detect multiple issues in a piece of problematic code at the same time.

### ISSUE THREAD

**Issue by Zlaojie** [mgechev/revive#1206]
**Describe the bug**
I wrote a piece of code. It has the following issues:
1. context.Context should be the first parameter of a function.
2. maximum number of arguments per function exceeded; max 2 but got 3.
3. error should be the last type when returning multiple items

And I used the revive configuration in golangci-lint to detect issues with this code. But it can only detect “maximum number of arguments per function exceeded; max 2 but got 3.”

**To Reproduce**
1. I am using version 1.63.4 of golangci-lint
2. And only use revive to detect my code

The .golangci.yml: 
```toml
linters:
  disable-all: true
  enable:
  - revive
linters-settings:
  revive:
    rules:
    - name: context-as-argument
      severity: error
      disabled: false
      exclude:
      - ''
    - name: error-return
      severity: warning
      disabled: false
      exclude:
      - ''
    - name: argument-limit
      severity: warning
      disabled: false
      exclude:
        - ''
      arguments: 2
```

The code:
```
func processData(data string,ctx context.Context, param5 []int) (err error, result string) {
	if data == "" {
		return "", errors.New("data cannot be empty")
	}

	select {
	case <-ctx.Done():
		return ctx.Err(), ""
	default:
		result = "Processed: " + data
		return nil, result
	}
}
```

**Expected behavior**
I hope revive can simultaneously detect all issues in the problematic code, rather than just one type of issue.

**Logs**
Command: golangci-lint run /Users/test/golangci-lint/index.go --timeout 60m --config=/Users/test/config/.golangci.yml
Results: 
```
/Users/test/golangci-lint/index.go:57:1: argument-limit: maximum number of arguments per function exceeded; max 2 but got 3 (revive)
func processData(data string,ctx context.Context, param5 []int) (result string, err error) {
```

**Comment by chavacava** [comment]
Hi @Zlaojie thanks for reporting the problem.
I was not able to reproduce the problem with the last version of `revive`:

```
$ revive problematic-code.go
problematic-code.go:3:1: error should be the last type when returning multiple items
problematic-code.go:3:30: context.Context should be the first parameter of a function
problematic-code.go:3:51: parameter 'param5' seems to be unused, consider removing or renaming it as _
problematic-code.go:1:1: should have a package comment
```

where `problematic-code.go` is
```go
package main

func processData(data string,ctx context.Context, param5 []int) (err error, result string) {
	if data == "" {
		return "", errors.New("data cannot be empty")
	}

	select {
	case <-ctx.Done():
		return ctx.Err(), ""
	default:
		result = "Processed: " + data
		return nil, result
	}
}
```

Please check if on your side you can reproduce by calling `revive` directly (without passing through `golang-ci`.

**Comment by Zlaojie** [comment]
> Hi [@Zlaojie](https://github.com/Zlaojie) thanks for reporting the problem. I was not able to reproduce the problem with the last version of `revive`:
> 
> ```
> $ revive problematic-code.go
> problematic-code.go:3:1: error should be the last type when returning multiple items
> problematic-code.go:3:30: context.Context should be the first parameter of a function
> problematic-code.go:3:51: parameter 'param5' seems to be unused, consider removing or renaming it as _
> problematic-code.go:1:1: should have a package comment
> ```
> 
> where `problematic-code.go` is
> 
> package main
> 
> func processData(data string,ctx context.Context, param5 []int) (err error, result string) {
> 	if data == "" {
> 		return "", errors.New("data cannot be empty")
> 	}
> 
> 	select {
> 	case <-ctx.Done():
> 		return ctx.Err(), ""
> 	default:
> 		result = "Processed: " + data
> 		return nil, result
> 	}
> }
> Please check if on your side you can reproduce by calling `revive` directly (without passing through `golang-ci`.

Oh Yes!!! I can  reproduce by calling `revive` directly. Why is there a problem with using golangci-lint?

```shell
$ revive /Users/test/golangci-lint/index.go
/Users/test/golangci-lint/index.go:57:1: maximum number of arguments per function exceeded; max 1 but got 2
/Users/test/golangci-lint/index.go:57:30: context.Context should be the first parameter of a function
/Users/test/golangci-lint/index.go:57:1: error should be the last type when returning multiple items
```

**Comment by chavacava** [comment]
I have not too much experience with `golangci-lint` therefore I can not help you in finding the reason errors reported by `revive` do not show in the `golangci-lint` result.
I suggest you to open an issue in the `golangci-lint` github issue tracker (share your `golangci-lint` configuration in the ticket)
Since `revive` seems to be working as expected, I will close this issue here. Feel free to reopen it if your investigations point `revive` as a suspect.
